src/librustdoc/html/render/print_item.rs

```diff
@@ -944,13 +944,17 @@ fn item_union(w: &mut Buffer, cx: &Context<'_>, it: &clean::Item, s: &clean::Uni
 fn print_tuple_struct_fields(w: &mut Buffer, cx: &Context<'_>, s: &[clean::Item]) {
     for (i, ty) in s
         .iter()
-        .map(|f| if let clean::StructFieldItem(ref ty) = *f.kind { ty } else { unreachable!() })
+        .map(|f| if let clean::StructFieldItem(ref ty) = *f.kind { Some(ty) } else { None })
```

Comment:
It should be possible at this stage, it means that the `clean::Item` was badly created... Can you find out where a non-`StructFieldItem` was added please?

Reply:
Do you mean it shouldn't be possible to create a `clean::Item` with `StructFieldItem` kind?

Reply:
I meant that a tuple item (when creating a new one) should only have `StructFieldItem` and discard everything else. (In `clean/mod.rs` iirc, sorry, not on computer at the moment)

Reply:
The kind we end up with is

```
StrippedItem(StructFieldItem(Tuple([])))
```

my brief look into how the render works suggests to me that this is correct, and that it just needs to be handled.

Reply:
Isn't there the same issue with:

```rust
pub enum Foo {
    Bar {
        x: u32,
        #[doc(hidden)]
        y: u8,
    }
}
```

Reply:
That does not panic

![image](https://user-images.githubusercontent.com/1502855/132108072-c12cfa06-a70d-4484-b65f-739c19f3bb75.png)

```rust
pub struct Foo(#[doc(hidden)] ());
```
does not panic either, and outputs 

![image](https://user-images.githubusercontent.com/1502855/132108162-38be6973-8777-4ec6-a546-3db5c596318a.png)

Which gives me the idea to do the same `_` for stripped/hidden tuple field